src/MCPClient/lib/clientScripts/parse_dataverse_mets.py

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import json
+import os
+import uuid
+
+# databaseFunctions requires Django to be set up
+import django
+django.setup()
+from django.utils import timezone
+
+# archivematicaCommon
+from archivematicaFunctions import get_file_checksum
+from custom_handlers import get_script_logger
+import databaseFunctions
+from main.models import Agent, File
+import metsrw
+
+logger = get_script_logger("archivematica.mcp.client.parse_dataverse_mets")
+transfer_objects_directory = "%transferDirectory%objects"
+
+
+def get_db_objects(job, mets, transfer_uuid):
+    """
+    Get DB objects for files in METS.
+
+    This also validates that files exist for each file asserted in the
+    structMap
+
+    :param mets: Parse METS file
+    :return: Dict where key is the FSEntry and value is the DB object
+    """
+    mapping = {}
+    for entry in mets.all_files():
+        if entry.type == 'Directory' or entry.label == 'dataset.json':
+            continue
+        # TODO remove this once RData files are returned
+        if entry.path.endswith('.RData'):
+            continue
+        # Retrieve the item name from the database. The proof-of-concept for
+        # Dataverse extracts objects from a bundle (zip) which can then be
+        # found on the root path of the transfer (the objects folder), but
+        # that means the initial lookup of the file might not resolve as
+        # anticipated as the directory structure of the bundle is reflected in
+        # the original path. We try the base name for the item below.
+        f = None
+        try:
+            item_path = os.path.join(transfer_objects_directory, entry.path)
+            logger.info(
+                "Looking for for file type: '{}' using relative path: {}"
+                .format(entry.type, item_path))
+            f = File.objects.get(originallocation=item_path,
+                                 transfer_id=transfer_uuid)
+        except File.DoesNotExist:
+            logger.info(
+                "Could not find file type: '{}' in the database: {}"
+                .format(entry.type, entry.path))
+        except File.MultipleObjectsReturned as e:
+            logger.info("Multiple entries for `%s` found. Exception: %s",
+                        entry.path, e)
```

Comment:
Here you're the optional keyword arguments to expand `%s`, this is good.
However, in L56, L70, L76 and maybe in other places you're using `str.format()` instead.